DUnitX.Attributes.pas

```diff
@@ -277,6 +277,19 @@   TestCaseAttribute = class(CustomTestCaseAttribute)
     property Values : TValueArray read GetValues;
   end;
 
+  /// <summary>
+  /// TestCaseProvider Attribute allows you, to pass a registered
+  /// Class to the Test, that provides the test function width the
+  /// needed data.
+  /// </summary>
+  TestCaseProviderAttribute = Class(TCustomAttribute)
+  protected
+     fname : string;
+  Public
+     Constructor Create(Const ProviderName:string);overload;
```

Comment:
Rather than provider name, can this not just be class reference?

Reply:
Well, its possible. In turn, depending on how many providers a user want's to use, you got a list of additional uses for the units of the providers in each fixture-unit. 
On big testprojects, this might be a disadvantage. 

Just an idea at the moment, its possible to do both ways. For the manager/factory it doesn't realy matter, if searching for a string or a class reference. This way, the user has the choice :)